src/jdk.compiler/share/classes/com/sun/tools/javac/code/Type.java

```diff
@@ -1151,7 +1151,11 @@ public void setEnclosingType(Type outer) {
 
         public List<Type> allparams() {
             if (allparams_field == null) {
-                allparams_field = getTypeArguments().prependList(getEnclosingType().allparams());
+                Type enclosingType = getEnclosingType();
+                if (!enclosingType.isUnbound() && enclosingType.getUpperBound() instanceof Type t) {
```

Comment:
Shouldn't this be a while loop in case you have G bounded by another type variable?

Reply:
You mean an intersection type? I think you are right. So do you have something like the following in mind?

```
static class Usage<T, G extends Getters<T> && SomethingElse> {
    public T test(G.Getter getter) {
        return getter.get(); // incompatible types: Object cannot be converted to T
    }
}
```

`G` would be raw if `SomethingElse` is raw if its definition is parametric, e.g. `SomethingElse<T>` and not raw if its definition is really `class SomethingElse`. Did you mean that?

Reply:
Nah, I was more thinking about
```
<T, U extends Getters<T>, G extends U>
```

Reply:
I think that wouldn't be a problem.

In the original example we try to type-check `G.Getter`. 
With your example `<T, U extends Getters<T>, G extends U>` we would type check `G.Getter` for validity and then `U.Getter`. But maybe I am wrong. 

In any case, both cases deserve corresponding test cases and correct treatment. Thanks @liach. Will update.

Reply:
Maybe I am wrong. We will figure it with more tests. Your comment was very useful I think. Thank you!!